### ISSUE TITLE
[rustdoc] Suggest trailing `\` instead of two spaces for hard line break

### ISSUE THREAD

**Issue by Marcono1234** [rust-lang/rust-clippy#12163]
### What it does

When a rustdoc comment contains two trailing spaces for a Markdown hard line break, the lint should suggest using a trailing backslash (`\`) instead.

rustdoc uses CommonMark, and it allows using `\` for hard line breaks, see the [specification](https://spec.commonmark.org/0.30/#hard-line-breaks).

### Advantage

A trailing `\` makes the intention for a hard line break clearer than two trailing spaces, which can be difficult to notice in Git diffs, and might be removed accidentally during refactoring.

### Drawbacks

- The choice whether to use two spaces or a `\` is subjective
(this could be solved by adding the lint to `clippy::pedantic`)
- If there is not actually a need for a hard line break (e.g. if a `> ...` follows), then the `\` might be shown literally ([CommonMark example](https://spec.commonmark.org/dingus/?text=foo%5C%0A%3E%20baz%0A))

### Example

```rust
/// First line··
/// Second line
fn foo() {}
```
(`·` indicating a space)

Could be written as:

```rust
/// First line\
/// Second line
fn foo() {}
```

**Comment by Jacherr** [comment]
This would probably end up being a restriction lint rather than being applied by default, but nevertheless sounds quite useful - I haven't seen the double space method used before, in all honesty.

@rustbot claim
